encodings/roaring/src/boolean/stats.rs

```diff
@@ -0,0 +1,159 @@
+use std::collections::HashMap;
+
+use croaring::Bitset;
+use vortex::stats::{ArrayStatisticsCompute, Stat, StatsSet};
+use vortex_error::{vortex_err, VortexResult};
+
+use crate::RoaringBoolArray;
+
+impl ArrayStatisticsCompute for RoaringBoolArray {
+    fn compute_statistics(&self, stat: Stat) -> VortexResult<StatsSet> {
+        if self.is_empty() {
+            return Ok(StatsSet::new());
+        }
+
+        // Only needs to compute IsSorted, IsStrictSorted and RunCount all other stats have been populated on construction
+        let bitmap = self.bitmap();
+        BitmapStats(
+            bitmap
+                .to_bitset()
+                .ok_or_else(|| vortex_err!("Bitmap to Bitset conversion run out of memory"))?,
+            self.len(),
+            bitmap.statistics().cardinality,
+        )
+        .compute_statistics(stat)
+    }
+}
+
+struct BitmapStats(Bitset, usize, u64);
+
+impl ArrayStatisticsCompute for BitmapStats {
+    fn compute_statistics(&self, _stat: Stat) -> VortexResult<StatsSet> {
+        let bitset_slice = self.0.as_slice();
+        // This is a weird case where the bitset is full of false values
+        // sometimes it will not allocate any underlying storage
+        if bitset_slice.is_empty() {
+            return Ok(StatsSet::from(HashMap::from([
+                (Stat::IsSorted, true.into()),
+                (Stat::IsStrictSorted, (self.1 == 1).into()),
+                (Stat::RunCount, 1.into()),
+            ])));
+        }
+
+        let whole_chunks = self.1 / 64;
+        let last_chunk_len = self.1 % 64;
+        let fist_bool = bitset_slice[0] & 1 == 1;
+        let mut stats = RoaringBoolStatsAccumulator::new(fist_bool);
+        for bits64 in bitset_slice[0..whole_chunks].iter() {
+            stats.next(*bits64);
+        }
+        stats.next_up_to_length(bitset_slice[whole_chunks], last_chunk_len);
+        Ok(stats.finish(self.2))
+    }
+}
+
+struct RoaringBoolStatsAccumulator {
+    last: bool,
+    is_sorted: bool,
+    run_count: usize,
+    len: usize,
+}
+
+impl RoaringBoolStatsAccumulator {
+    fn new(first_value: bool) -> Self {
+        Self {
+            last: first_value,
+            is_sorted: true,
+            run_count: 1,
+            len: 0,
+        }
+    }
+
+    pub fn next_up_to_length(&mut self, next: u64, len: usize) {
+        assert!(len <= 64);
+        self.len += len;
+        for i in 0..len {
+            let current = ((next >> i) & 1) == 1;
+            if !current & self.last {
```

Comment:
&&

Reply:
Also: maybe worth brief one sentence comment to clarify bool sort order

Reply:
This is one of those cases where linter is trying to be too clever